src/IsCountable.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+namespace Zend\Validator;
+
+use Zend\Validator\Exception\RuntimeException;
+
+class IsCountable extends AbstractValidator
+{
+    const NOT_COUNTABLE = 'notCountable';
+    const NOT_EQUALS    = 'notEquals';
+    const GREATER_THAN  = 'greaterThan';
+    const LESS_THAN     = 'lessThan';
+
+    /**
+     * Validation failure message template definitions
+     *
+     * @var array
+     */
+    protected $messageTemplates = [
+        self::NOT_COUNTABLE => "The input must be an array or an instance of \\Countable",
+        self::NOT_EQUALS    => "The input count must equal '%count%'",
+        self::GREATER_THAN  => "The input count must be less than '%max%', inclusively",
+        self::LESS_THAN     => "The input count must be greater than '%min%', inclusively",
+    ];
+
+    /**
+     * Additional variables available for validation failure messages
+     *
+     * @var array
+     */
+    protected $messageVariables = [
+        'count' => ['options' => 'count'],
+        'min'   => ['options' => 'min'],
+        'max'   => ['options' => 'max'],
+    ];
+
+    /**
+     * Options for the between validator
+     *
+     * @var array
+     */
+    protected $options = [
+        'count' => null,
+        'min'   => null,
+        'max'   => null,
+    ];
+
+    /**
+     * Returns true if and only if $value is countable (and the count validates against optional values).
+     *
+     * @param  string $value
+     * @return bool
+     * @throws Exception\RuntimeException
+     */
+    public function isValid($value)
+    {
+        if (! (is_array($value) || $value instanceof \Countable)) {
+            throw new RuntimeException($this->messageTemplates[self::NOT_COUNTABLE]);
+        }
+
+        $count = count($value);
+
+        if (is_numeric($this->options['count'])) {
```

Comment:
Also, it looks like the validator will have different behavior based on whether `count`, `min`, or `max` are set; in fact, it appears that these should be _mutually exclusive_. As such, I wonder if the constructor and/or `setOptions()` should have some logic to prevent cases where one is already set/discovered, and an attempt is made to set a different option?